src/pykeen/triples/triples_factory.py

```diff
@@ -648,6 +761,21 @@ def relations_to_ids(self, relations: Union[Collection[int], Collection[str]]) -
             if not isinstance(e, int):
                 raise ValueError(f"{self.__class__.__name__} cannot convert relation IDs from {type(e)} to int.")
         return cast(Collection[int], relations)
+    
+    def timestamps_to_ids(self, timestamps: Union[Collection[int], Collection[str]]) -> Collection[int]:
+        """Normalize timestamps to IDs.
+
+        :param timestamps: A collection of either integer identifiers for timestamps or
+            string labels for timestamps (that will get auto-converted)
+        :returns: Integer identifiers for timestamps
+        :raises ValueError: If the ``timestamps`` passed are string labels
+            and this triples factory does not have a timestamp label to identifier mapping
+            (e.g., it's just a base :class:`CoreTriplesFactory` instance)
+        """
+        for e in timestamps:
+            if not isinstance(e, int):
+                raise ValueError(f"{self.__class__.__name__} cannot convert relation IDs from {type(e)} to int.")
```

Comment:
Here, you need to allow passing `YYYY-MM-DD` format from above, too (that is the purpose of this method - to normalize labels *or* ids to ids).